### ISSUE TITLE
Using real CD for copy protected software?

### ISSUE THREAD

**Issue by Olddantrucker** [86Box/86Box#1907]
**Is your feature request related to a problem? Please describe.**
Some games require a real CD or you get a 'Please Insert Correct CD' after installing then trying to launch said game.  IE: Need for Speed High Stakes,Midtown Madness II which PCEM allows you to use your HOST drive which I can play it via that way.  On PCEM with Win 98 I can pop the CD in my real drive and it will recognize it after a reboot of it (PCEM).  

**Describe the solution you'd like**
An icon similar to PCEM of 'Use Host CD' .

**Describe alternatives you've considered**
Making ISO/Bin Cu images of the same CD but copy protection is still a problem!!!

**Additional context**

**Comment by TC1995** [comment]
You can use UltraISO with its convert option to convert it to BIN/CUE without losing copy protected data. (Not Save As directly which makes it lose the protected data).

**Comment by Olddantrucker** [comment]
> You can use UltraISO with its convert option to convert it to BIN/CUE without losing copy protected data. (Not Save As directly which makes it lose the protected data).

I was using Any To Iso a lightweight program that makes the images just fine only no copy protected games work so I fail  to see how UltraISO will be any different?

**Comment by TC1995** [comment]
UltraISO worked for me for copy protected stuff in BIN/CUE format.

**Comment by Olddantrucker** [comment]
> UltraISO worked for me for copy protected stuff in BIN/CUE format.

I'm giving it a try with Midtown Madness II.  Can it convert from my real copy? If so do I do it to Bin.Cu NOT an ISO format? What happens if I covert it to ISO instead?

**Comment by Olddantrucker** [comment]
Edit: It seems it doesn't do anything. I press convert and place it in my 86box but the button seems to act as fi it's not there. Could it be due to using a trial version or something else entirely?

**Comment by TC1995** [comment]
You need a cracked version of it.

**Comment by Olddantrucker** [comment]
Sorry didn't work for me. I mean the Bin.Cue image made it but it still asked me to insert the correct CD even though I installed it okay on 86box. I have to resort to using the HOST CD feature on PCEM to play it so for those games I'm sticking to PCEM.  Too bad Sarah went weird and quit. Not sure what happened there but I guess too much lockdowns and government tyranny gets the best of us!

**Comment by Olddantrucker** [comment]
> You need a cracked version of it.

Just as I suspected your an NPC trying to sell something or send a virus thru. Good thing I didn't fall for that! I almost did but know better now.

**Comment by TC1995** [comment]
What? I have a cracked version of UltraISO just fine.

**Comment by Olddantrucker** [comment]
> What? I have a cracked version of UltraISO just fine. Also, "you're" instead of "your" in this case.

All the 'cracked' versions lead to a single zip file saying 'password is' which the virus scanner stops it form completing and NO I AM NOT Compromising my PC by turning it off especially giving the fact that all the sites lead to the same zip file.

**Comment by Olddantrucker** [comment]
> UltraISO worked for me for copy protected stuff in BIN/CUE format.

I got the CUE file for it and it still wouldn't let me play without using the actual CD so I am back to PCEM for that regards. PCEM still kicks 86box in the box..........when it comes to protected games due to the REAL CD feature.

**Comment by Diicorp95** [comment]
> The only thing PCEM lacks is sound is always quiet unlike 86Box which has a good volume control.

PCem have volume controls!

**Comment by OBattler** [comment]
I removed the old code because it was a total mess in the end. Eventually, I plan to reintroduce host CD-ROM drive support, but it takes time to even just port it due to the code divergence.

Though, I will point out that I'm seriously disappointed at how this whole matter has been handled from our side. Not ONE person offered themselves to do this thing, instead, we only had TC1995 telling the reporter to "just use a cracked version" of the game. That's *NOT* how you handle such issues at all.

**Comment by OBattler** [comment]
But, let me give a more detailed run down about the problems with real CD-ROM pass-through, at least on Windows: the problem is emulating command 0xBE (READ CD) and its relatives (such as READ CD MSF). The Windows IOCTL API unfortunately does not have the same range of format selections as the 0xBE command. So the last time, I ended up using SPTD, but that has a different problem - if the drive is not using PIO, you have no way to know the exact amount of data transferred, and I have to know that when I'm emulating the PIO version of this command. So the last time, I ended up writing a complex set of functions that detected the mode of the selected sector by sending mode-specific 0xBE commands to the drive until one returned no error. But that's very slow.

Yes, PCem may handle your game fine, but it would still fail on a lot of other stuff. And there's no easy way to improve that code.

A better solution may be implementing support for MDS+MDF but there's no human-readable description of the format anywhere, and there's also no library suitable for use with an emulator.

But regardless of which of the two we decide to do, we're chronically understaffed right now, so it's going to take a while.
